src/Serilog/Capturing/PropertyValueConverter.cs

```diff
@@ -169,6 +185,61 @@ LogEventPropertyValue CreatePropertyValue(object? value, Destructuring destructu
         return new ScalarValue(value.ToString() ?? "");
     }
 
+    Destructuring? GetCachedFallbackDestructuring(
+#if !NET5_0
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+#endif
+        Type type)
+    {
+        if (_fallbackDestructuringCache.TryGetValue(type, out var fallbackDestructuring)) return fallbackDestructuring;
+
+        var fallback = GetFallbackDestructuring(type);
+        _fallbackDestructuringCache[type] = fallback;
+
+        return fallback;
+    }
+
+    Destructuring? GetFallbackDestructuring(
+#if !NET5_0
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+#endif
+        Type type)
+    {
+        // Check direct hit:
+        var destructuringFallback = GetFallbackDestructuring(type, true);
+        if (destructuringFallback != null) return destructuringFallback;
+
+        // Check interfaces:
+        foreach (var @interface in type.GetInterfaces())
```

Comment:
This will return all interfaces in the full hierarchy, including those from base types.